### PR TITLE
Migrate docs from wiki to github

### PR DESCRIPTION
**Pull request by sladyn98**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,434 @@
 Jenkins LDAP Plugin
 ===================
 
-Read more: [https://wiki.jenkins-ci.org/display/JENKINS/LDAP+Plugin](https://wiki.jenkins-ci.org/display/JENKINS/LDAP+Plugin)
+*Note*: This plugin was part of the Jenkins core until 1.468. After
+that, it was split out into a separately-updateable plugin. However, for
+backwards compatibility purposes, subsequent core releases still bundle
+it. If you do not use this plugin at all, you can simply disable it.
+
+# Description
```
Comment (timja) [inline README.md]: There should be only one top level heading on a page, in this case the heading can be removed

File: README.md [hunk 2]
```diff
@@ -1,7 +1,434 @@
...
+resolution and user lookup and typically surface for users as issues
+with API token or Jenkins CLI access but can also appear with features
+such as the [Authorize Project
+plugin](https://wiki.jenkins.io/display/JENKINS/Authorize+Project+plugin)
+or other plugins that require details of user permissions or group
+membership outside of a user's web session)
+
+ 
+
+# Configuration
```
Comment (timja) [inline README.md]: Adjust the heading level to one level deeper

File: README.md (hunk 1 above)
Comment (MarkEWaite) [inline README.md]:
```suggestion
## Description
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,434 @@
...
+`ou=people`
+
+### User search filter
+
+One of the searches Jenkins does on LDAP is to locate the user record
+given the user name.
+
+ This field determines the query to be run to identify the user record.
+
+The query is almost always `uid={0`} as per defined in RFC 2798, so in
```
Comment (MarkEWaite) [inline README.md]: Formatting error in original?
```suggestion
The query is almost always `uid={0}` as per defined in RFC 2798, so in
```

File: README.md [hunk 4]
```diff
@@ -1,7 +1,434 @@
...
+most cases you should leave this field empty and let this default kick
+in.
+
+If your LDAP server doesn't have `uid` or doesn't use a meaningful `uid`
+value, try `mail={0`}, which lets people login by their e-mail address.
```
Comment (MarkEWaite) [inline README.md]:
```suggestion
value, try `mail={0}`, which lets people login by their e-mail address.
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,434 @@
...
+
+If you do specify a different query, specify an LDAP query string with
+marker token {`0`}, which is to be replaced by the user name string
```
Comment (MarkEWaite) [inline README.md]:
```suggestion
marker token `{0}`, which is to be replaced by the user name string
```

File: README.md [hunk 6]
```diff
@@ -1,7 +1,434 @@
...
+field, such as: `(member={0})`
+
+The LDAP server may be able to use query hints to optimize the search.
+For example:
+
+-   If all the groups you are interested in are within a specific
+    subtree, adding the subtree information to the filter should improve
+    performance.
+
+&nbsp;
```
Comment (MarkEWaite) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -1,7 +1,434 @@
...
+In such cases enabling caching may improve performance of Jenkins with
+the risk of delayed propagation of user changes from LDAP and increased
+memory usage on the master.
+
+Note: The default configuration is to leave the cache turned off.
+
+### Environment Properties
+
+As of 1.7 of the LDAP plugin, you can now specify additional Environment
+Properties to provide the backing Java LDAP client API. See [Oracle's
```
Comment (MarkEWaite) [inline README.md]:
```suggestion
properties to provide the backing Java LDAP client API. See [Oracle's
```

File: README.md [hunk 8]
```diff
@@ -1,7 +1,434 @@
...
+documentation](http://docs.oracle.com/javase/7/docs/technotes/guides/jndi/jndi-ldap.html){.external-link} for
```
Comment (MarkEWaite) [inline README.md]: Use Java 8 reference documentation.  Java 7 no longer supported with Jenkins
```suggestion
documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/jndi/jndi-ldap.html){.external-link} for
```

File: README.md [hunk 9]
```diff
@@ -1,7 +1,434 @@
...
+|-------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `com.sun.jndi.ldap.connect.timeout` | This is the socket connection timeout in milliseconds. If your LDAP servers are all close to your Jenkins server you can probably set a small value, e.g. 5000 milliseconds. Setting a value smaller that this may result in excessive timeouts due to the TCP/IP connection establishment retry mechanism. |
+| `com.sun.jndi.ldap.read.timeout`    | This is the socket read timeout in milliseconds. If your LDAP queries are all fast you can probably set a low value. The value is ignored if the Jenkins Master is running on Java 1.5. A reasonable default is 60000 milliseconds.                                                                         |
```
Comment (MarkEWaite) [inline README.md]: Jenkins no longer supports Java 1.5.  Remove the disclaimer.
```suggestion
| `com.sun.jndi.ldap.read.timeout`    | This is the socket read timeout in milliseconds. If your LDAP queries are all fast you can probably set a low value. A reasonable default is 60000 milliseconds.                                                                         |
```

File: README.md (hunk 8 above)
Comment (sladyn98) [reply README.md]: Should I also remove the `{external.link}`

Comment (timja) [reply README.md]: yes

File: README.md [hunk 10]
```diff
@@ -1,7 +1,432 @@
...
+membership outside of a user's web session)
+
+ 
+
+## Configuration
+
+Select LDAP for the Security Realm. You will most likely need to
+configure some of the Advanced options. There is on-line help available
+for each option. 
+![](docs/images/Screen_Shot_2017-05-02_at_09.52.20.png){.confluence-embedded-image}
```
Comment (timja) [inline README.md]:
```suggestion
![](/docs/images/Screen_Shot_2017-05-02_at_09.52.20.png)
```

File: README.md [hunk 11]
```diff
@@ -1,7 +1,432 @@
...
+### Test LDAP Settings
+
+This button will allow you to check the full LDAP configuration settings
+which you have defined (as compared with the field validation which only
+verifies a subset of the configuration)
+
+Clicking this button will display a modal dialog to prompt you to
+provide a username and password:
+
+![](docs/images/test-ldap-dialog.png){.confluence-embedded-image}
```
Comment (timja) [inline README.md]:
```suggestion
![](/docs/images/test-ldap-dialog.png)
```

File: README.md [hunk 12]
```diff
@@ -1,7 +1,432 @@
...
+    resolve their accounts by using their username and an empty password
+    =\> *You do not want to lock legitimate users out*
+-   In most cases, you will be using LDAP groups, so ensure that you
+    verify reverse group lookup by testing with a user account that is a
+    member of at least one group (do not forget the empty password trick
+    to perform lookup). If there are important groups for your Jenkins
+    instance, try using at least one user for each important group =\>
+    You want to ensure that group lookup functions correctly.
+
+![(warning)](docs/images/warning.svg){.emoticon
```
Comment (timja) [inline README.md]:
```suggestion
![(warning)](/docs/images/warning.svg)
```

possibly enough to just use ⚠️

File: README.md [hunk 13]
```diff
@@ -1,7 +1,432 @@
...
+.emoticon-warning} **NOTE** it is quite likely that existing
```
Comment (timja) [inline README.md]:
```suggestion
**NOTE** it is quite likely that existing
```

File: README.md [hunk 14]
```diff
@@ -1,7 +1,432 @@
...
+There is an extension point for providing a strategy to resolve the LDAP
+groups that the user belongs to. There are two implementations provided
+in the LDAP plugin:
+
+-   Search for groups containing user (default)
+-   Parse user attribute for list of groups
+
+#### Search for groups containing user
+
+![](docs/images/Screen_Shot_2014-07-15_at_10.19.23.png){.confluence-embedded-image}
```
Comment (timja) [inline README.md]:
```suggestion
![](/docs/images/Screen_Shot_2014-07-15_at_10.19.23.png)
```

File: README.md [hunk 15]
```diff
@@ -1,7 +1,432 @@
...
+    if using Active Directory's matching rule in chain extension, e.g.
+    `(&(objectCategory=group)(member:1.2.840.113556.1.4.1941:={0}))`
+
+Note: in this field there are two available substitutions:  
+{`0`} - the fully qualified DN of the user  
+{`1`} - the username portion of the user
+
+#### Parse user attribute for list of groups
+
+![](docs/images/Screen_Shot_2014-07-15_at_10.19.14.png){.confluence-embedded-image}
```
Comment (timja) [inline README.md]: I would consider renaming the file as well
```suggestion
![](/docs/images/Screen_Shot_2014-07-15_at_10.19.14.png)
```

File: README.md [hunk 16]
```diff
@@ -1,7 +1,432 @@
...
+| `com.sun.jndi.ldap.read.timeout`    | This is the socket read timeout in milliseconds. If your LDAP queries are all fast you can probably set a low value. A reasonable default is 60000 milliseconds.                                                                         |
+
+## Troubleshooting
+
+The following [Groovy
+script](https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+Script+Console)
+can be useful when trying to determine whether you have group search
+configured correctly:
+
+``` syntaxhighlighter-pre
```
Comment (timja) [inline README.md]:
```suggestion
```java
```

**Comment by sladyn98** [comment]
Shouldn't the exporter itself remove these `embedded` links.

**Comment by timja** [comment]
> Shouldn't the exporter itself remove these `embedded` links.

Yes file an issue here https://github.com/jenkins-infra/jenkins-wiki-exporter / send a PR happy to review it 😄

File: docs/images/warning.svg [hunk 17]
```diff
@@ -0,0 +1 @@
+<svg width="16" height="16" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="M9.756 1.723l5.627 10.32A2 2 0 0 1 13.627 15H2.373a2 2 0 0 1-1.756-2.957l5.627-10.32a2 2 0 0 1 3.512 0z" fill="#FFC400"/><path d="M9 8.8c0 .662-.448 1.2-1 1.2s-1-.538-1-1.2V5.2C7 4.538 7.448 4 8 4s1 .538 1 1.2v3.6zM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2z" fill="#253858"/></g></svg>
```
Comment (timja) [inline docs/images/warning.svg]: this appears to be unused now?

Comment (sladyn98) [reply docs/images/warning.svg]: Yeah it was there since the start, guess we can get rid of it now, since we make use of the warning sign by github

**Comment by sladyn98** [comment]
> @sladyn98 I think the CI failure was a transient failure. If you'll commit another change, however minor it may be, that should allow the CI job to evaluate it again. Approved once the CI job passes

Thanks for the review mark will push a minor commit soon  :)

**Pull request merged**